### DISCUSSION TITLE
UTF-8 BOM character breaks workflow parser

### DISCUSSION THREAD

**Discussion by dogboydog** [community/community#25914] [Actions]
<p>Hi all,</p>
<p>Loving GitHub actions and the new workflow_dispatch functionality! I just wanted to raise this issue which caused me some confusion. I now have a workaround detailed in the link at the bottom, but wanted to know the staff and other users’ thoughts on this.</p>
<p>Here is an example patch that will break the parser. It seems to cause unpredictable error messages later in the workflow file (“you have a syntax error at line 15”, etc.)</p>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/dogboydog/manual-workflow-test/blob/084e6017a349363d88303a39542e921a01b59dbe/failed_parse_patch.patch" target="_blank" rel="nofollow noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/dogboydog/manual-workflow-test/blob/084e6017a349363d88303a39542e921a01b59dbe/failed_parse_patch.patch" target="_blank" rel="nofollow noopener">dogboydog/manual-workflow-test/blob/084e6017a349363d88303a39542e921a01b59dbe/failed_parse_patch.patch</a></h4>
<pre><code class="lang-patch">Index: .github/workflows/quick_build.yaml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
&lt;+&gt;UTF-8
===================================================================
--- .github/workflows/quick_build.yaml	(revision 976bf688492ab5c161b78bed7282b2d4085c3f18)
+++ .github/workflows/quick_build.yaml	(revision a5762b6b4f2940a46be1b1ee3b87b32d6e54023b)
@@ -1,12 +1,11 @@
-name: Single build and upload
+﻿name: Single build and upload
 
 on:
   workflow_dispatch:
     inputs:
-      logLevel:
-        description: 'Log level'
+      gitRef:
+        description: 'Enter the git ref (branch, commit hash, tag, etc.) to build.'
         required: true
-        default: 'warning'
</code></pre>

  This file has been truncated. <a href="https://github.com/dogboydog/manual-workflow-test/blob/084e6017a349363d88303a39542e921a01b59dbe/failed_parse_patch.patch" target="_blank" rel="nofollow noopener">show original</a>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I would link to some failed runs but it seems like they disappeared after I stripped out the character myself and GitHub was able to parse the workflow name.</p>
<p>What do you think of this issue? Should the parser strip it out, or is there at least some way the parser can detect it’s presence and give a specific message about it. It seems like it could be safely removed. I know you are not expecting this character but some editors automatically insert it.</p>
<p>There is more information here but I figured it is no longer exactly my initial issue from that thread, so I made a new one. Please forgive any frustration I expressed there as I debugged this issue. Thank you</p>
<blockquote>
<p><a href="https://github.community/t/cant-trigger-workflow-manually/121740/36" class="inline-onebox">Can't trigger workflow manually</a></p>
</blockquote>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/dogboydog">@dogboydog</a>,</p>
<p>I forked your repository and create a new yaml following your patch file. However, i <code>cannot</code> reproduce the error on my workflow.<br>
For example: <a href="https://github.com/weide-zhou/manual-workflow-test/actions/runs/188160858/workflow" rel="nofollow noopener">https://github.com/weide-zhou/manual-workflow-test/actions/runs/188160858/workflow</a></p>
<p>I’m not sure 100% follow, let me know if there are any misunderstanding.</p>
<p>The <code>input</code> value will not be added to env, hence, you cannot echo it with <code>$</code> or <code>$INPUT_</code>, please get it from context, like <code>${{ github.event.inputs.logLevel }}</code>.</p>
<p>Thanks</p>

**Comment by dogboydog** [comment]
<p>Hmm odd. There are screenshots of the error in my other thread but I’ll post them here.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/d5286f3cd714df2ec40e743cb4ef2486e239be54.jpeg" data-download-href="/uploads/short-url/upGkzk0E01Gp2sGG0V7uKCTor64.jpeg?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/9920963/181094084-e4e7d493-cb83-42c7-9cac-1bcefa20c060.jpeg" alt="image" data-base62-sha1="upGkzk0E01Gp2sGG0V7uKCTor64" width="690" height="279" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/d/d5286f3cd714df2ec40e743cb4ef2486e239be54_2_690x279.jpeg, https://aws1.discourse-cdn.com/github/optimized/2X/d/d5286f3cd714df2ec40e743cb4ef2486e239be54_2_1035x418.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/2X/d/d5286f3cd714df2ec40e743cb4ef2486e239be54.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/d5286f3cd714df2ec40e743cb4ef2486e239be54_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1175×476 73.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/c0c8b65e4f04dea894f6dc1d7952138685fdb41e.png" data-download-href="/uploads/short-url/rvrHjsfBexDHR65rSKaNT0sjfmC.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/9920963/181094085-38b60e3d-0f04-4710-9eae-917d517ec898.png" alt="image" data-base62-sha1="rvrHjsfBexDHR65rSKaNT0sjfmC" width="690" height="351" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/c0c8b65e4f04dea894f6dc1d7952138685fdb41e_2_690x351.png, https://aws1.discourse-cdn.com/github/original/2X/c/c0c8b65e4f04dea894f6dc1d7952138685fdb41e.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/c/c0c8b65e4f04dea894f6dc1d7952138685fdb41e.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/c0c8b65e4f04dea894f6dc1d7952138685fdb41e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">840×428 49.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/77aabfbf9298d287e164bbeae7f4f093f50e00d7.png" data-download-href="/uploads/short-url/h4CE7jHQYilkBrRdTs47yBMHiLB.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/9920963/181094086-261c285e-4f3d-4e87-a803-8ba48e01b39e.png" alt="image" data-base62-sha1="h4CE7jHQYilkBrRdTs47yBMHiLB" width="690" height="460" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/7/77aabfbf9298d287e164bbeae7f4f093f50e00d7_2_690x460.png, https://aws1.discourse-cdn.com/github/optimized/2X/7/77aabfbf9298d287e164bbeae7f4f093f50e00d7_2_1035x690.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/7/77aabfbf9298d287e164bbeae7f4f093f50e00d7_2_1380x920.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/77aabfbf9298d287e164bbeae7f4f093f50e00d7_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">2199×1466 199 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/4a0b5df218baf965ac401ecf5213bdc9b9d920a7.png" data-download-href="/uploads/short-url/az1GyHcR4SXUJjycSnmpuRDhXmv.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/9920963/181094087-12272abe-3ce1-4c8f-97ce-78ed95ebeabf.png" alt="image" data-base62-sha1="az1GyHcR4SXUJjycSnmpuRDhXmv" width="690" height="240" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/4/4a0b5df218baf965ac401ecf5213bdc9b9d920a7_2_690x240.png, https://aws1.discourse-cdn.com/github/optimized/2X/4/4a0b5df218baf965ac401ecf5213bdc9b9d920a7_2_1035x360.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/4/4a0b5df218baf965ac401ecf5213bdc9b9d920a7_2_1380x480.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/4a0b5df218baf965ac401ecf5213bdc9b9d920a7_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">2305×803 134 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Also this pull request from staff from a previous thread involving this issue before I figured out the exact character made the same fix which made my workflow start parsing correctly again:<br>
<a href="https://github.com/dogboydog/manual-workflow-test/pull/1/files" rel="nofollow noopener">https://github.com/dogboydog/manual-workflow-test/pull/1/files</a> But I encountered the issue several times after this PR from creating other workflows.</p>
<p>Is it possible GitHub rolled out a fix for this in the last few days? <a class="mention" href="/cschleiden">@cschleiden</a>?  I can’t get it to happen anymore either, so that’s a good thing. Is there a changelog somewhere that shows bug fixes? I can only find actions to <em>create</em> changelogs by searching.</p>
<p>And yeah I know about the input thing. If you’re talking about test_manual2.yaml, that is an example from my other topic that demonstrates that the environmental variable is not set.  But thank you</p>
<p>Edit:  I think they might have just rolled out a fix. I’m happy they did! I missed this message in another topic and didn’t realize they were already tracking this issue. So, sorry if this is a duplicate.</p>
<blockquote>
<p>And yes, that seems to be a bug in our parser, we should report that error. I’ll file an issue for us to take a look :bowing_man:‍♂</p>
</blockquote>

**Comment by cschleiden** [comment, marked as answer]
<p><a class="mention" href="/dogboydog">@dogboydog</a> Thanks for the detailed report. I fixed an issue a bit more than a week ago where we ignore the BOM characters. Let me know if you can still repro this somewhere, happy to take another look :slight_smile:</p>

**Comment by dogboydog** [comment]
<p>Thanks for replying. I was wrestling with this issue on Sunday the 26th still (as soon as I stripped the BOM out, it starting parsing again), but I just tried to reproduce and I can’t anymore. If you fixed it more than a week ago I’m not sure why I was still seeing it on Sunday, but it seems all good now.  Seems totally fixed. Thank you</p>

**Comment by paulbizsunny** [comment]
<p>Good morning dear, you guys are working hard to make the platform friendly keep it up thank you</p>
